component/stream/stream_cache.go

```diff
@@ -190,17 +147,26 @@ func (c *cache) teardown() {
 	log.Trace("streamcache:: tearing down stream cache")
 	c.Lock()
 	defer c.Unlock()
-	for fileKey := range c.files {
-		c.files[fileKey].fileBlockBuffer.Purge()
-		delete(c.files, fileKey)
+	for _, block := range c.blocks.Keys(false) {
+		bk := block.(blockKey)
+		c.blocks.Remove(bk)
+		c.evictedBlock.handle.DataBuffer.Remove(c.evictedBlock)
+		c.evictedBlock = blockKey{}
 	}
-
 	// Cleanup block residing on disk path
 	if c.diskPersistence {
 		c.wipeoutDiskCache()
 	}
 }
 
+func (c *cache) removeCachedHandle(handle *handlemap.Handle) {
+	c.Lock()
```

Comment:
we can have handle level lock instead of entire cache policy lock here.

Reply:
same as above. One thing to note - perf did not change much when I removed this lock.

Reply:
@tasherif-msft can you add a comment in the relevant areas to mention that we cant remove this lock due to potential deadlock issues? Just for future code maintainability?